lib/spack/spack/stage.py

```diff
@@ -480,7 +480,12 @@ def print_errors(errors):
         else:
             print_errors(errors)
 
-            err_msg = 'All fetchers failed for {0}'.format(self.name)
+            # Simplify the error name by removing the prefix when present
+            name = self.name if not self.name.startswith(stage_prefix) else \
+                self.name.replace(stage_prefix, '')
+            err = 'Manual download is required' if manual_download else \
+                'All fetchers failed'
+            err_msg = '{0} for {1}'.format(err, name)
```

Comment:
What do you think if, instead of passing a Boolean like `manual_download`, we permit packages to pass a custom error message (a string)? If I am not missing something that would permit, without a lot of changes in the current logic, to have packages that can point you at how to obtain the software in the download error message etc.

Reply:
Changed to prefer a provided fetch error message over the default.